cpp/refactor_module/algorithm/refactor.cpp

```diff
@@ -460,9 +469,80 @@ void Refactor::CollapseNode(mgp_list *args, mgp_graph *memgraph_graph, mgp_resul
           return;
         }
       }
-    } 
-      
-    
+    }
+
+  } catch (const std::exception &e) {
+    record_factory.SetErrorMessage(e.what());
+    return;
+  }
+}
+
+namespace {
+
+template <typename node_or_rel>
+void NormalizeToBoolean(node_or_rel object, std::string &property_key, std::unordered_set<mgp::Value> &true_values,
```

Comment:
Also, I think this can be passed as &, because here we don't need to make a copy as this object exists somewhere else.

Reply:
In function parse object.ValueNode() is sent which can't be sent with reference since it creates a new object :(

Reply:
And this function is always called with either object.ValueNode() or object.ValueRelationship()